src/api.php

```diff
@@ -0,0 +1,819 @@
+<?php
+/**
+ * Public API.
```

Comment:
@szepeviktor what's the idea for this file? Are we moving all of this content to the proper classes and this code to `compat/`?

I'm asking because it's not very wise to have an 800-line file with all the functions dumped inside.

Reply:
You have already agreed to this in CU.

Reply:
Please tell me your decision!

Reply:
APIs via classes, moving these functions to `compat`